### PR TITLE
Added a Python script that converts a CF standards XML file to a YAML file.

### PR DESCRIPTION
**Pull request by jeff-cohere**
This Python script can be used as part of a tool chain for downloading the CF conventions and generating a YAML file that we can use to accept/reject long field names and aliases for quantities that have counterparts in CF.

The resulting YAML file is the first of two "dictionaries" that we need. The second is one of our own making, that contains field names and aliases for quantities that we track but that do not appear in the CF conventions. It also establishes a format for these dictionaries.

I encourage anyone to try it out! Just download the CF Conventions XML file (using the URL included in the script's front matter ) and run the script on it.

```
wget http://cfconventions.org/Data/cf-standard-names/76/src/cf-standard-name-table.xml
scripts/cf-xml-to-yaml cf-standard-name-table.xml
```

I've also added a `build_cf_dictionary` target that does this for you. Just type `make build_cf_dictionary` in your build directory, and your scream/data source directory will be populated with the generated YAML file.

Addresses #699

### CODE REVIEW

**Review by AaronDonahue — APPROVED** [review]
Looks good to me.  And it would be great to have an easy way to generate the YAML file if a new CF XML comes online.

File: components/scream/scripts/cf-xml-to-yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env python
+
+"""
+Given an XML file containing the CF conventions for standardized field names
+(https://cfconventions.org/standard-names.html), this tool generates a YAML
+file with the same information.
+
+Usage: cf-xml-toyaml <cf-xml-file>
```
Comment (AaronDonahue) [inline components/scream/scripts/cf-xml-to-yaml:8]: nitpicky, but should the text read `cf-xml-to-yaml`?  Notice the missing hyphen.

Comment (jeff-cohere) [reply components/scream/scripts/cf-xml-to-yaml]: Whoops! Nice catch.

**Review by PeterCaldwell — APPROVED** [review]
looks good!

**Comment by jeff-cohere** [comment]
Updated with a new build target, so I'm re-requesting approval. I think that ties off this PR.

**Comment by PeterCaldwell** [comment]
The new version still looks good. My only question is why you put the new target in scream/components/scream/src/share/. I'm not sure where would be better, but I wouldn't think to look here to build a tool...

**Comment by jeff-cohere** [comment]
> The new version still looks good. My only question is why you put the new target in scream/components/scream/src/share/. I'm not sure where would be better, but I wouldn't think to look here to build a tool...

The real "tool" is the script in the scripts/ directory. The target is just  CMake logic and can live in any `CMakeLists.txt` file. I guess I thought it should belong with the atmosphere driver code, since that's where we care about validating field names.

**Review by PeterCaldwell — APPROVED** [review]
Ok, so long as you thought about target location, I'm cool with it.

**Pull request merged**